tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md

```diff
@@ -0,0 +1,364 @@
+
+Open a terminal, and run the following commands. The first one is to install git:
+
+``` bash
+sudo apt install git
+```
+
+Now, clone your repository to your local workspace, using the address displayed in GitHub. Something like this, but with your username instead of mine:
+
+
+``` bash
+mkdir -p ~/workspace
+cd workspace
+git clone https://github.com/elopio/hello-snap-username.git
+cd hello-snap-username
+```
+
+To create a snap project, open a text editor, and write the following content:
+
+``` yaml
+name: hello-yourname
+version: '2.10'
+summary: GNU Hello, the "hello world" snap
+description: GNU hello prints a friendly greeting.
+grade: stable
+confinement: strict
+
+apps:
+  hello:
+    command: bin/hello
+
+parts:
+  gnu-hello:
+    source: http://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
+    plugin: autotools
+```
+
+Save it to `~/workspace/hello-snap-username/snap/snapcraft.yaml` (remember to replace "username" with yours!)
+
+This is the packaging metadata of a very simple snap, that, as the description says, will just print a greeting. The “[Create your first snap]” tutorial can provide details for all the lines of this file, so please refer to it to know more about the packaging. In here we will be focusing just on the continuous delivery, so let’s go on and push this snap to our GitHub repository:
+
+
+``` bash
+git add snap/snapcraft.yaml
+git commit -m "add the hello snap"
+git push
+```
+
+We now have a GitHub project containing packaging details for a snap, it’s time to look at Travis CI...
+
+## Setting up Travis CI
+Duration: 2:00
+
+To use Travis CI, first we will have to sign up there. Let’s go to [https://travis-ci.org/auth] and click "Sign in with GitHub". Then, click "Authorize application" to allow Travis to access your GitHub account.
+
+Now, enable the GitHub repository of your snap by flipping the switch next to it at [https://travis-ci.org/profile/]
+
+![](images/3.png)
+
+Now, we need a `.travis.yml` file for your project. Travis CI will run the steps specified in this file on every pull request and every time a change is pushed to the master branch. These steps are going to be a few bash commands and scripts.
+
+First, create a file named `.travis.yml` in your project folder, with the following content:
+
+``` bash
+language: bash
+```
+
+Later you can use Travis CI to run your tests. For that you will have to adjust this file depending on the language of your project, and add the install and script steps. But for now, we don’t need any of that as we’ll just make continuous delivery to the store.
+
+In the next step we’ll make sure that your snap can be built successfully in Travis CI, so we can push the generated snap to the store.
+
+## Building the snap in a clean environment
+Duration: 5:00
+
+Travis CI executors are using Ubuntu 14.04, but are missing some kernel changes to be able to run snapcraft as a snap itself. Since they support Docker, we will use a snapcraft Docker container to build the snap.
```

Comment:
I would say:

Travis CI uses a build environment based on Ubuntu 14.04, which is missing some requirements for running snapcraft. However, it does have support for Docker, so we will use a Docker container to build the snap